index.bs

```diff
@@ -364,58 +369,62 @@ to ease the use of Suborigins in legacy applications are also presented. See
-the suborigin namespace `profile`, this would be serialized as
+For example, a resource hosted at `https://example.com/` in
+the suborigin namespace `profile` would be serialized as
 `https://profile_example.com/`.
 
-Similarly, if a resource is hosted at `https://example.com:8080/` in
-the suborigin namespace `separate`, this would be serialized as
+Similarly, a resource hosted at `https://example.com:8080/` in
+the suborigin namespace `separate` would be serialized as
 `https://separate_example.com:8080/`.
 
-Internally, the <a>suborigin namespace</a> must be tracked by the user agent.
-When the origin needs to be serialized for a resource, the user agent should
+Internally, the user agent just track the <a>suborigin namespace</a> of the resource.
+When the origin needs to be serialized, the user agent should
 follow the algorithm in [[#serializing]].
 
+Note that the underscore character is not a valid character for hostnames used
+on the web [[!RFC1123]] and as a result, the serialization above cannot collide
+with existing hostnames.
+
 ## Opting into a Suborigin ## {#opting-in}
 
 Unlike the `sandbox` attribute, suborigin namespaces are predictable and
 controllable. Because of this, potentially untrusted content cannot opt into
 suborigins, unlike iframe sandboxes. If they could, then an XSS on a site could
 enter a specific suborigin and access all of its resources, thus violating the
-privilege separation suborigins are intended to protect. To prevent this, the
-server (rather than a resource itself) is treated as the only authoritative
-source of the suborigin namespace of a resource. This is implemented through an
-additional header-only Content Security Policy directive `suborigin`, which
-takes a string value that is the namespace. For example, to put a resource in
-the `testing` suborigin namespace, the server would specify the following
-directive in the CSP header:
+isolation suborigins intend to provide. To prevent this, the
+server (rather than a resource itself) is the only authoritative
+source of the suborigin namespace of a resource. The server communicates the
+suborigin of a resource to the user agent through a new `suborigin` header,
+which takes a string value that is the namespace. For example, to put a
+resource in the `testing` suborigin namespace, the server would specify the
+following HTTP header in the response:
 
 <pre>
   suborigin: testing
 </pre>
 
-## The `suborigin` Directive ## {#the-suborigin-directive}
+## The `suborigin` header ## {#the-suborigin-header}
 
-Suborigins are defined by a <dfn>suborigin directive</dfn> in the Content
-Security Policy [[!CSP2]] of the resource. The syntax for the name and value of
-the directive are described by the following ABNF grammar:
+Suborigins are defined by a <dfn>suborigin</dfn> HTTP response header. The syntax
+for the name and value of the header are described by the following ABNF
+grammar (RFC5234):
```

Comment:
Please change "(RFC5234)" to a normative link to the RFC (I think "[[!RFC5234]]" should do the trick).